lib/models.js

```diff
@@ -683,7 +709,7 @@ class Cardinality {
 
     if (this._history && this._history.length > 0) {
       out["history"] = this._history.map(h => ({
-        "source": h["source"],
+        "source": h._source.fqn,
```

Comment:
This will crash if `source` is `null` or `undefined`.  Do we need to protect against that, or is that an impossible scenario?